content/post/building-docker-directly-on-a-raspberry-pi.md

```diff
@@ -0,0 +1,244 @@
+* enabled swap space (1 GByte of memory is not sufficient!)
+* Docker Engine 1.11
+
+
+#### Installing the Build Environment
+
+As you will see, setting up the build environment is damn easy, because we'll just
+use the latest release of HypriotOS which has the required Docker Engine 1.11.1
+already pre-installed. These steps here are done on OS X, on a Linux or Windows
+box it's a little bit different.
+
+*Step 1:* install HypriotOS 0.8.2 on a SD card
+
+```
+$ flash https://github.com/hypriot/image-builder-rpi/releases/download/v0.8.2/hypriotos-rpi-v0.8.2.img.zip
+```
+Now boot the Raspberry Pi 3 with the newly flashed SD card.
+
+*Step 2:* deploy SSH keys
+
+(Note: login credentials are username=pirate, password=hypriot)
+
+First, wait until the Raspberry Pi is booted up and is visible on the network
+to gather it's IP address.
+```
+$ ping -c 1 black-pearl.local
+PING black-pearl.local (192.168.2.113): 56 data bytes
+64 bytes from 192.168.2.113: icmp_seq=0 ttl=64 time=5.697 ms
+```
+Second, deploy your standard SSH keys to the Raspberry Pi.
+```
+$ ssh-add
+$ ssh-keygen -R 192.168.2.113
+$ ssh-copy-id pirate@192.168.2.113
+```
+
+Now we're able to login into the Raspberry Pi without using a password.
+```
+$ ssh pirate@192.168.2.113
+
+$ uname -a
+Linux black-pearl 4.4.15-hypriotos-v7+ #1 SMP PREEMPT Mon Jul 25 08:46:52 UTC 2016 armv7l GNU/Linux
+```
+
+*Step 3:* install some build dependencies
+
+```
+$ sudo apt-get update
+$ sudo apt-get install -y make
+```
+
+*Step 4:* using a swap file
+
+This will create and use a default 2x 1GByte swap file at /var/swap.
+
+```
+$ sudo apt-get install -y dphys-swapfile
+$ ls -alh /var/swap
+-rw------- 1 root root 1.8G Jul 30 17:58 /var/swap
+```
+
+If we need a larger swap file, we could edit the config file `/etc/dphys-swapfile`,
+but for our use case the default 1.8 GByte swap file is sufficient.
+
+
+### Clone the Docker repo
+
+```
+$ cd
+$ git clone https://github.com/docker/docker.git
+$ cd ~/docker
+$ git checkout v1.12.0
+```
+As you can see, we're using a tagged release for Docker to build this exact version.
+
+### How to apply some extra Pull Requests
+
+Just to be honest, right now the build process for Docker v1.12.0 will fail on ARM.
+This is a known issue and will be fixed hopefully soon. But luckily there is already
+a [pull request](https://github.com/docker/docker/pull/25192)
+which enables us to build the Docker .deb packages. This gives me
+the change to explain how you can easily apply a PR on top of a release version.
+
+But first we have to tell `git` a few required settings about my username and
+email address. Otherwise we'll get an error message when applying the PR.
+```
+$ git config --global user.email "dieter.reuter@me.com"
+$ git config --global user.name "Dieter Reuter"
+```
+
+So, let's cherry pick this specific pull request.
+```
+$ curl -sSL https://patch-diff.githubusercontent.com/raw/docker/docker/pull/25192.patch | git am
```

Comment:
OK, this seems to be a cleaner way. Will try it now, and then change the blog post.